frontend/src/metabase/query_builder/components/DataSelector/DataSelector.jsx

```diff
@@ -428,7 +427,10 @@ export class UnconnectedDataSelector extends Component {
     }
   }
 
-  isLoadingDatasets = () => this.props.loading;
+  isSearchLoading = () => {
+    const { models, metrics, allLoading } = this.props;
+    return models == null || metrics == null || allLoading;
```

Comment:
This is an old Entity Framework bug - initially `loading` or `allLoading` is set to `false` and data is `undefined`.

<img width="602" alt="Screenshot 2024-11-28 at 15 48 47" src="https://github.com/user-attachments/assets/86b428e2-bdff-4546-946d-905f55b1175b">

Reply:
Until this got fixed I had race conditions in e2e tests that sometimes it thought that there are no models when opened initially.